IPython/utils/_process_posix.py

```diff
@@ -59,11 +61,18 @@ class ProcessHandler(object):
 
     @property
     def sh(self):
-        if self._sh is None:        
```

Comment:
I repeat it below, but we should only set self._sh once, not on every call, so this line should be restored.